### ISSUE TITLE
SIMD subgroup meeting on 2022-02-18

### ISSUE THREAD

**Issue by ngzhian** [WebAssembly/relaxed-simd#50]
This is a joint meeting with flexible-vectors and any SIMD related presentations are also welcome, e.g. new architectures or toolchains, longer term discussion that don't fit into flexible-vectors or relaxed-simd.

The meeting will be on Friday, Feb 18 at 9:00AM - 10:00AM PDT/ 5:00PM - 6:00PM CET.

If this meeting doesn't already appear on your calendar, or you are a new attendee, please fill out this [form](https://forms.gle/9eB2ZYaziPEcTJabA) to attend.

**Comment by lars-t-hansen** [comment]
It would be useful to discuss the state of relaxed simd, what the final outcome of the fpenv discussion was, what work remains in terms of spec and overview text, and who is going to do it.  We might discuss whether there are more instructions to be added or whether we are ready to push this forward to Phase 3, as Firefox has a full implementation of the current spec modulo fpenv.  Apparently LLVM and V8 both have implementations too.  @yurydelendik says that he has time to take on some of this work.

**Comment by ngzhian** [comment]
> It would be useful to discuss the state of relaxed simd, what the final outcome of the fpenv discussion was, what work remains in terms of spec and overview text, and who is going to do it.

The outcome of the fpenv discussion is in the meeting notes for the last presentation we did to the CG on 2021-11-09: https://github.com/WebAssembly/meetings/blob/main/main/2021/CG-11-09.md#update-on-relaxed-simd-fpenv-discussions-and-poll-for-phase-2-zhi-an-ng-15-min

There is some weirdness in spec text (as currently proposed), for example consistency across operators, we probably want the min and max to behave similarly, and fms/fma to be similar, we need a way to say that (that's what I recall and what I can tell from the meeting notes, but I might be missing something).

I have not yet tweaked the spec text, and will be able to get to it soon ~2 weeks. I welcome any help (make a PR to https://github.com/ngzhian/relaxed-simd)

> as Firefox has a full implementation of the current spec modulo fpenv. Apparently LLVM and V8 both have implementations too. 

Nice! Yes V8 has prototype implementation on x64 and arm64, LLVM and related tools have support for experimenting via `-mrelaxed-simd`.

fpenv won't require any engine support IMO, the whole point of this new fpenv is to make it a spec construct.

For phase 3, we need
- test suite that runs on engines, so that means we need tests, and spec updated to understand this
- finalize instruction names #42 
- finalize binary encoding #51 
- @Maratyszcza has a couple more instructions in mind

**Comment by dtig** [comment]
Another possible agenda item would be to discuss open issues that have not been merged into the spec, and the likelihood of them to be merged into the overview, so implementers can prototype in parallel. Specifically https://github.com/WebAssembly/relaxed-simd/issues/40, https://github.com/WebAssembly/relaxed-simd/issues/17 for now, but there are other issues we could triage through depending on how much time we have. 

Separately, @Maratyszcza brought up approximation instructions, and whether this proposal should handle them today, and that could probably also be a good item to discuss in the meeting tomorrow.

**Comment by ngzhian** [comment]
Next meeting issue https://github.com/WebAssembly/relaxed-simd/issues/56
